src/sentry/eventstream/snuba.py

```diff
@@ -161,6 +161,27 @@ def insert(
         event_type = self._get_event_type(event)
         occurrence_data = self._get_occurrence_data(event)
 
+        # instead of normalizing and doing custom 'contexts' processing in snuba, we elect to do it here instead to
+        # avoid having to clutter up snuba code with business logic
+        if event_type == EventStreamEventType.Generic:
+            updated = False
+            if event_data.get("contexts"):
+                contexts = event_data.get("contexts", {})
```

Comment:
Was going back and forth about whether to put the data processing logic here(right before sending it out through the stream) or somewhere more upstream like in the ingest consumer before we store the event in nodestore. If we decide to do more mutations, the event in nodestore could potentially look a lot different than what we store in the search_issues dataset.

Reply:
I think it makes sense to do it here. Mirrors what we already do for errors, since the context extraction already happens post eventstream.